samples/javascript_nodejs/16.proactive-messages/bots/proactiveBot.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityHandler, TurnContext } = require('botbuilder');
+
+
+class ProactiveBot extends ActivityHandler {
+    constructor(conversationReferences) {
+        super();
+
+        // Dependency injected dictionary for storing ConversationReference objects used in NotifyController to proactively message users
+        this.conversationReferences = conversationReferences;
+
+        this.onConversationUpdate(async (context, next) => {
+
+           this.addConversationReference(context.activity);
+            
+            await next();
+        });
+
+        this.onMembersAdded(async (context, next) => {
+            const membersAdded = context.activity.membersAdded;
+            for (let cnt = 0; cnt < membersAdded.length; cnt++) {
+                if (membersAdded[cnt].id !== context.activity.recipient.id) {
+                    const welcomeMessage = "Welcome to the Proactive Bot sample.  Navigate to http://localhost:3978/api/notify to proactively message everyone who has previously messaged this bot.";
+                    await context.sendActivity(welcomeMessage);
+                }
+            }
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+
+        this.onMessage(async (context, next) => {
+            this.addConversationReference(context.activity);
+
+            // Echo back what the user said
+            await context.sendActivity(`You sent '${context.activity.text}'`);
+            await next();
+        });
+    }
+
+    addConversationReference(activity) {
+        const conversationReference = TurnContext.getConversationReference(activity);
+        this.conversationReferences[conversationReference.conversation.id] = conversationReference;
+    }
+}
```

Comment:
This looks good but can you make sure to run the linter and push another commit. it looks like the spacing would have been picked up by the linter.

Reply:
Just ran the linter and pushed another commit.